### PR TITLE
Supports multiple instantiation of DRAMSim3 module

### PR DESCRIPTION
**Pull request by mrutt92**
Adds support for multiple DRAMSim3 instantiations.

### CODE REVIEW

**Comment by mrutt92** [comment]
@tommydcjung can you try this out and let me know if it fixes your problem? thanks!

**Review by tommydcjung — COMMENTED** [review]
I think this fixes the issue. I still need to run more tests, since the simulation takes really long time with multiple dramsim3

File: bsg_test/bsg_dramsim3.cpp [hunk 1]
```diff
@@ -142,92 +267,26 @@ extern "C" void bsg_dramsim3_exit(void)
  * @param[in] config_p       The path to the configuration file for the memory system (.ini)
  * @return true if the system was succesfully initialized, otherwise false.
```
[the remark this answers is not in the document]
Comment (mrutt92) [reply bsg_test/bsg_dramsim3.cpp]: thanks

**Comment by mrutt92** [comment]
Cool - thank you.

**Comment by taylor-bsg** [comment]
Hi Tommy, what do you mean by “the simulation takes a really long time with multiple DRAMSim”? Is there a reason it should take a really long time?

**Comment by tommydcjung** [comment]
It could be a combination of things. Running 2048 core simulation by itself takes a long time. 4 instantiation of dramsim3 allocates 32GB of dram memory for each simulation. So running in parallel could take a long time.

**Review by tommydcjung — APPROVED** [review]

**Comment by taylor-bsg** [comment]
Trying to determine if DRAMSim itself imposes a tax or whether it is purely
the # of cores..

**Comment by mrutt92** [comment]
What's the status on this? Can it be merged?

**Comment by mrutt92** [comment]
@taylor-bsg unless this has broken anything, or made a single instance slower, can we merge so it doesn't go stale? It's been sitting in a holding pattern for over two weeks now.

**Comment by tommydcjung** [comment]
We can merge it after we test it with cuda-lite? I think I have tested it sufficiently from my end.

**Comment by mrutt92** [comment]
It's been tested with cuda-lite. Everything runs just fine.

**Pull request merged**